File: src/ProfileGenerator.java
public class ProfileGenerator {
    
    private double distance, desiredCruiseVelocity, maxAcceleration;

    public ProfileGenerator(double distance, double cruiseVelocity, double accleration){
        this.distance = distance;
        desiredCruiseVelocity = cruiseVelocity;
        maxAcceleration = accleration;
    }
    public static void main(String[] args) throws Exception {
        
        ProfileGenerator pg = new ProfileGenerator(12, 3, 3);
        pg.run();
    }

    public void run(){

        double maxVelocity = maxAcceleration * Math.sqrt(distance / maxAcceleration);

        double v_cruise = Math.min(maxVelocity, desiredCruiseVelocity);

        double timeToAccel = v_cruise/maxAcceleration;
        double x1 = v_cruise * timeToAccel;   // distance covered while acceleratinng and decelerating
        double x2 = distance - x1;                      // distance covered at cruise velocity 
        double t_cruise = x2/v_cruise;                      // time at cruise velocity
        double loopRunTime_ms = 20;
        double totalRunTime = timeToAccel *2 + t_cruise;
        
        
        int runningTime_ms = 0;
        double currVelocity, currAcceleration, currDistance;
        currVelocity = currAcceleration = currDistance = 0;



        while(runningTime_ms <= totalRunTime *1000){

            if(runningTime_ms < timeToAccel * 1000){
                currAcceleration = maxAcceleration;
            }else if(runningTime_ms < (timeToAccel + t_cruise) * 1000){
                currAcceleration = 0;
            }else{
                currAcceleration = -1 * maxAcceleration;
            }

            currVelocity += currAcceleration * loopRunTime_ms /1000.0;

            currVelocity = Math.min(currVelocity, v_cruise);
            currVelocity = Math.max(currVelocity, 0);

            currDistance += currVelocity * loopRunTime_ms/ 1000.0 + .5 * currAcceleration * Math.pow(loopRunTime_ms/1000.0, 2);

            System.out.printf("( %4d %.4f %.4f %.4f) \n", runningTime_ms, currDistance, currVelocity, currAcceleration);

            runningTime_ms += loopRunTime_ms;

        
        }

        System.out.println(totalRunTime);


  /*      for(runningTime_ms = 0; runningTime_ms <= timeToAccel*1000; runningTime_ms += 20){
        
            currAcceleration = maxAcceleration;
            currVelocity = maxAcceleration * runningTime_ms /1000.0;
            currDistance = .5 * maxAcceleration * Math.pow(runningTime_ms/1000.0, 2);

            System.out.printf("( %4d %.4f %.4f %.4f) \n", runningTime_ms, currDistance, currVelocity, currAcceleration);

        }

        while(runningTime_ms <= (timeToAccel + t_cruise)*1000){
            currAcceleration = 0;
            currVelocity = v_cruise;
            currDistance += v_cruise * loopRunTime_ms/1000.0; 

            System.out.printf("( %4d %.4f %.4f %.4f) \n", runningTime_ms, currDistance, currVelocity, currAcceleration);

            runningTime_ms += 20;

        }

        while(runningTime_ms <= totalRunTime*1000){

            currAcceleration = -1 * maxAcceleration;
            currVelocity += currAcceleration * loopRunTime_ms /1000.0;
            currDistance += currVelocity * loopRunTime_ms/ 1000.0 + .5 * maxAcceleration * Math.pow(loopRunTime_ms/1000.0, 2);

            System.out.printf("( %4d %.4f %.4f %.4f) \n", runningTime_ms, currDistance, currVelocity, currAcceleration);

            runningTime_ms +=20;
        }
*/


    }
}

File: src/ProfileGenerator.py

import matplotlib.pyplot as plt
import math


distance =  5.13 #float(input("Distance: "))
desired_velocity =  2 #float(input("Cruise Velocity: "))
acceleration = 2 # float(input("Acceleration: " ))




def profileGenerator(distance, cruise_velocity, acceleration):
   
    max_velocity = math.sqrt(distance * acceleration)
    cruise_velocity = min(max_velocity, cruise_velocity)
    
    time_to_accel = cruise_velocity/ acceleration
    distance_after_accel = .5 * acceleration * math.pow(time_to_accel,2)
    cruise_distance = distance - 2* distance_after_accel
    time_at_vcruise = cruise_distance/ cruise_velocity

    running_time_ms = 0
    loop_run_time = .02
    current_pos = 0
    current_vel = 0
    current_accel = 0

    t_list = []
    x_list = []
    v_list = []
    a_list = []

    while(running_time_ms <= (2*time_to_accel + time_at_vcruise)*1000):

        if(running_time_ms < time_to_accel * 1000):
            current_accel = acceleration
        elif(running_time_ms < (time_to_accel + time_at_vcruise)*1000):
            current_accel = 0
        else:
            current_accel = -1 * acceleration
    
        current_vel += current_accel * loop_run_time
        current_vel = min(cruise_velocity, current_vel)

        current_pos += current_vel* loop_run_time + .5 * current_accel * loop_run_time * loop_run_time

        print("t: {:.2f} x: {:.2f} v:{:.2f} a: {:.2f}".format(running_time_ms/1000,current_pos,current_vel,current_accel))

        t_list.append(running_time_ms/1000)
        x_list.append(current_pos)
        v_list.append(current_vel)
        a_list.append(current_accel)

        running_time_ms += 1000*loop_run_time
    

    plt.plot(t_list, x_list, label= 'Position')
    plt.plot(t_list, v_list, label = 'Velocity')
    plt.plot(t_list, a_list, label = 'Acceleration')
    plt.xlabel('Time, s', fontsize = 12)
    plt.legend()




profileGenerator(distance, desired_velocity, acceleration)
plt.show()